README.md
# group-github-activity-readme

监听某人某些仓库的活动合集，demo 示例：

<!--START_SECTION:activity-->
## yiliang114 in [microsoft/vscode](https://github.com/microsoft/vscode)
### Issue List: 
1. [gulp task do not display task name #189556](https://github.com/microsoft/vscode/issues/189556) still in Open ❗ at 2023-08-03 
2. [vscode.dev Cannot open a file in a folder #187826](https://github.com/microsoft/vscode/issues/187826) had been Close 🔒 at 2023-07-13 
3. [Language pack files for some extensions not found #187795](https://github.com/microsoft/vscode/issues/187795) had been Close 🔒 at 2023-07-13 
4. [Automatically expand the last expanded folder when the page is refreshed in web. #187788](https://github.com/microsoft/vscode/issues/187788) had been Close 🔒 at 2023-07-13 
5. [CustomBuiltinExtensions FromLocations will not automatically load the correct `package.nls.{locale}.json` #186701](https://github.com/microsoft/vscode/issues/186701) had been Close 🔒 at 2023-06-30 
6. [Explorer refresh does not show latest data of the file system provider #185312](https://github.com/microsoft/vscode/issues/185312) had been Close 🔒 at 2023-06-16 
7. [Editor does not restore focus after exiting command panel #183890](https://github.com/microsoft/vscode/issues/183890) had been Close 🔒 at 2023-05-31 
8. [Some Unit Test Problems #183506](https://github.com/microsoft/vscode/issues/183506) still in Open ❗ at 2023-05-26 
9. [The Toggle Command of Webview can only open, but not close panel #182266](https://github.com/microsoft/vscode/issues/182266) still in Open ❗ at 2023-05-12 
10. [Forward and backward of trackpad in Webview should be disabled? #182013](https://github.com/microsoft/vscode/issues/182013) still in Open ❗ at 2023-05-10 
11. [The button is squeezed in Extension Preview page #178474](https://github.com/microsoft/vscode/issues/178474) had been Close 🔒 at 2023-03-28 
12. [Search results were not restored when the window was reloaded. #177658](https://github.com/microsoft/vscode/issues/177658) had been Close 🔒 at 2023-03-20 
13. [Failed to open workspace settings file under empty workspace #177639](https://github.com/microsoft/vscode/issues/177639) had been Close 🔒 at 2023-03-20 
14. [Find Widget border-bottom-radius Supplement #176789](https://github.com/microsoft/vscode/issues/176789) had been Close 🔒 at 2023-03-10 
15. [Hard to launch a workbench without a workspace #176763](https://github.com/microsoft/vscode/issues/176763) had been Close 🔒 at 2023-03-10 
16. [Installation dependency is very slow in china for the first time #173242](https://github.com/microsoft/vscode/issues/173242) had been Close 🔒 at 2023-02-03 
17. [Leaving duplicate workspace nodes When refreshing #171458](https://github.com/microsoft/vscode/issues/171458) still in Open ❗ at 2023-01-17 
18. [Label name with `undefined` when the workspace is not named #171402](https://github.com/microsoft/vscode/issues/171402) had been Close 🔒 at 2023-01-16 
19. [Is there a plan for `vscode` sdk to support Dialog API? #171138](https://github.com/microsoft/vscode/issues/171138) had been Close 🔒 at 2023-01-12 
20. [Should `vscode.workspace.fs.stat` API  return all content providered by User-defined FileSystem provider #170539](https://github.com/microsoft/vscode/issues/170539) had been Close 🔒 at 2023-01-04 
21. [The translation of the delete file menu is inconsistent. #170087](https://github.com/microsoft/vscode/issues/170087) had been Close 🔒 at 2022-12-27 
22. [Extra scroll bar in terminal group #167752](https://github.com/microsoft/vscode/issues/167752) had been Close 🔒 at 2022-11-30 
23. [Regular expression search file is supported in many places. #167751](https://github.com/microsoft/vscode/issues/167751) had been Close 🔒 at 2022-11-30 
24. [Toggle Editor Layout should hidden when there is only one editor group #167745](https://github.com/microsoft/vscode/issues/167745) had been Close 🔒 at 2022-11-30 
25. [TreeView of Debug Watch Style Defect problem #167724](https://github.com/microsoft/vscode/issues/167724) still in Open ❗ at 2022-11-30 
26. [package.json `browser` field cannot be converted in `acquireWebNodePaths` #165162](https://github.com/microsoft/vscode/issues/165162) had been Close 🔒 at 2022-11-01 
27. [Unable to specify `folder-uri` run with `./scripts/code-web.sh` #164414](https://github.com/microsoft/vscode/issues/164414) had been Close 🔒 at 2022-10-24 
## yiliang114 in [microsoft/vscode-l10n](https://github.com/microsoft/vscode-l10n)
### Issue List: 
1. [Cannot export value on deconstruction #119](https://github.com/microsoft/vscode-l10n/issues/119) still in Open ❗ at 2023-07-07 
## yiliang114 in [microsoft/vscode-test-web](https://github.com/microsoft/vscode-test-web)
### Issue List: 
1. [How to start with https? #42](https://github.com/microsoft/vscode-test-web/issues/42) still in Open ❗ at 2022-09-21 
## yiliang114 in [coder/code-server](https://github.com/coder/code-server)
### Issue List: 
1. [[Bug]: Some link path errors #6187](https://github.com/coder/code-server/issues/6187) had been Close 🔒 at 2023-05-03 
## harbin1053020115 in [microsoft/vscode](https://github.com/microsoft/vscode)
### Issue List: 
1. [Add codelens to stickyScroll #182949](https://github.com/microsoft/vscode/issues/182949) had been Close 🔒 at 2023-05-19 
2. [vscode.dev get crashed while saving a untitled editor #177638](https://github.com/microsoft/vscode/issues/177638) had been Close 🔒 at 2023-03-20 
3. [There is a wrong comment word in 'IPCClient' class(From ipc.ts in vscode source code) #175197](https://github.com/microsoft/vscode/issues/175197) had been Close 🔒 at 2023-02-23 
4. [yarn run watch-web is not worked #168550](https://github.com/microsoft/vscode/issues/168550) had been Close 🔒 at 2022-12-09 
### PR List: 
### PR List: 
### PR List: 
<!--END_SECTION:activity-->

## 贡献者

<!-- readme: collaborators,contributors -start -->
<table>
<tr>
    <td align="center">
        <a href="https://github.com/yiliang114">
            <img src="https://avatars.githubusercontent.com/u/11473889?v=4" width="100;" alt="yiliang114"/>
            <br />
            <sub><b>易良</b></sub>
        </a>
    </td></tr>
</table>
<!-- readme: collaborators,contributors -end -->


CHANGELOG.md
# Changelog

## [1.2.7](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.6...v1.2.7) (2023-08-05)


### Bug Fixes

* PR List 模版渲染逻辑 ([9a93c5f](https://github.com/cloud-webide/group-github-activity-readme/commit/9a93c5fa07555eaf36e44de1e6173573874f99c2))

## [1.2.6](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.5...v1.2.6) (2023-08-05)


### Bug Fixes

* use changelogithub ([fd9e5ad](https://github.com/cloud-webide/group-github-activity-readme/commit/fd9e5ad4a58516e233dae1778a10fe02a89ef789))
* 其他项目在执行是，会找不到 ./index.js ([bf2f749](https://github.com/cloud-webide/group-github-activity-readme/commit/bf2f749749b9599b054979369073305d06cf30b1))

## [1.2.5](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.4...v1.2.5) (2023-08-05)


### Bug Fixes

* 删除多余依赖 & 修复 gh release  upload 构建产物路径 ([b1c5c79](https://github.com/cloud-webide/group-github-activity-readme/commit/b1c5c79b73c7fbfdf67a908a794da0ca7d50e6b6))

## [1.2.4](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.3...v1.2.4) (2023-08-05)


### Bug Fixes

* use changelogithub ([bd33a87](https://github.com/cloud-webide/group-github-activity-readme/commit/bd33a8731ae17aca50550124774692ddfde40b8e))
* use changelogithub ([40c0a09](https://github.com/cloud-webide/group-github-activity-readme/commit/40c0a09a26521fe1a42ea8da0b780ad108aae2a0))
* 删除多余换行符 & 删除空的目录模版 ([6b9a3d6](https://github.com/cloud-webide/group-github-activity-readme/commit/6b9a3d644bd17b09fffe7fffb0858728bf44fe62))
* 构建产物到根目录（便于  gh release upload 更新发布单文件产物） ([b6baa8b](https://github.com/cloud-webide/group-github-activity-readme/commit/b6baa8b017f7386cdcc3365eda0362f8c0acc8f8))

## [1.2.3](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.2...v1.2.3) (2023-08-05)


### Bug Fixes

* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([2e28a72](https://github.com/cloud-webide/group-github-activity-readme/commit/2e28a72af49a1e00f9eb19cd010a1f3bc42f9f37))

## [1.2.2](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.1...v1.2.2) (2023-08-05)


### Bug Fixes

* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([d6a6f79](https://github.com/cloud-webide/group-github-activity-readme/commit/d6a6f7941322efa224e5ec9017426e8983facc95))

## [1.2.1](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.2.0...v1.2.1) (2023-08-05)


### Bug Fixes

* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([eed6cdd](https://github.com/cloud-webide/group-github-activity-readme/commit/eed6cdde47c66896e4878672daf3fcb9f6b2136d))
* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([5ff2234](https://github.com/cloud-webide/group-github-activity-readme/commit/5ff22349e832bb0a10a26b3919d40a2cb544ffc4))
* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([9b9128a](https://github.com/cloud-webide/group-github-activity-readme/commit/9b9128adabd9ed65c5ad86898c593610c65f51f7))
* 使用 release-please-action 发布时，提交构建产物 Close [#15](https://github.com/cloud-webide/group-github-activity-readme/issues/15) ([3136f56](https://github.com/cloud-webide/group-github-activity-readme/commit/3136f5665dbd77b94e801300f751a9d0094d66a9))

## [1.2.0](https://github.com/cloud-webide/group-github-activity-readme/compare/v1.1.0...v1.2.0) (2023-08-05)


### Features

* 使用 @vercel/ncc 将 ts 编译产物打成一个文件（action 执行不需要安装依赖了） ([677c317](https://github.com/cloud-webide/group-github-activity-readme/commit/677c317860dec3b29542aeaa13a121b879d48bcc))
* 使用 @vercel/ncc 将 ts 编译产物打成一个文件（action 执行不需要安装依赖了） ([686bc34](https://github.com/cloud-webide/group-github-activity-readme/commit/686bc348eda4a0f403ddceed17b8c846df5a5d46))
* 修改活动内容模版 透出标题 ([0d999e6](https://github.com/cloud-webide/group-github-activity-readme/commit/0d999e636000c8596196c19a664936dd33018438))
* 修改活动内容模版 透出标题 ([2b4bd1c](https://github.com/cloud-webide/group-github-activity-readme/commit/2b4bd1c275a081336f238cde340c7092804a8ea6))
* 修改活动内容模版 透出标题 ([b915468](https://github.com/cloud-webide/group-github-activity-readme/commit/b915468449f19ad5614a08d5e5cefa2bbdab7c42))
* 修改活动内容模版 透出标题 ([508af85](https://github.com/cloud-webide/group-github-activity-readme/commit/508af857e0253b1631f11deb842faa86d1df952c))
* 修改活动内容模版 透出标题 ([3bc68ca](https://github.com/cloud-webide/group-github-activity-readme/commit/3bc68ca23b7ff884b4b0c54649db02dca863b90b))


### Bug Fixes

* Error: Cannot find module '@actions/core' Close [#3](https://github.com/cloud-webide/group-github-activity-readme/issues/3) ([6c91610](https://github.com/cloud-webide/group-github-activity-readme/commit/6c916109d893d7dc6c42a86490ef25d546d52606))
* 填充 action 默认值 ([7453138](https://github.com/cloud-webide/group-github-activity-readme/commit/745313862c70f5da704d15ffd59f89e9415cb253))
* 填充 action 默认值 ([6bb2252](https://github.com/cloud-webide/group-github-activity-readme/commit/6bb22521470407d3c6ca0a2be46264c9abba8e72))
* 多用户 action 处理 ([c19b415](https://github.com/cloud-webide/group-github-activity-readme/commit/c19b4158ca5a60026a68f45a524ad7de3e3062f7))
* 多用户 action 处理 ([0164cf1](https://github.com/cloud-webide/group-github-activity-readme/commit/0164cf1f986e821da1c2973ac1887edcc2c6bbf8))
* 恢复 &lt;!--START_SECTION:activity--&gt; ([6bd0159](https://github.com/cloud-webide/group-github-activity-readme/commit/6bd0159d73e3b0126e62b64b8fee04b2c2852464))
* 恢复 &lt;!--START_SECTION:activity--&gt; ([42bcb6c](https://github.com/cloud-webide/group-github-activity-readme/commit/42bcb6c4b9e6a995da27de5414bb0f6c4492903b))

src/a.js
const fs = require("fs");

const TARGET_FILE = "README.md";

(async (tools) => {
  const content = `## yiliang114's activities:

  ### [microsoft/vscode](https://github.com/microsoft/vscode)
  
  1. 💪 Opened PR [#187902](https://github.com/microsoft/vscode/pull/187902) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-07-14 06:11:21
  2. ❗ Opened issue [#187826](https://github.com/microsoft/vscode/issues/187826) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-07-13 12:36:02
  3. 💪 Opened PR [#187796](https://github.com/microsoft/vscode/pull/187796) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-07-13 06:25:23
  4. ❗ Opened issue [#187795](https://github.com/microsoft/vscode/issues/187795) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-07-13 06:23:44
  5. ❗ Opened issue [#187788](https://github.com/microsoft/vscode/issues/187788) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-07-13 02:47:35
  6. ❗ Opened issue [#186701](https://github.com/microsoft/vscode/issues/186701) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-06-30 08:08:20
  7. 💪 Opened PR [#186607](https://github.com/microsoft/vscode/pull/186607) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-06-29 08:53:16
  8. ❗ Opened issue [#185312](https://github.com/microsoft/vscode/issues/185312) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-06-16 08:30:41
  9. ❗ Opened issue [#183890](https://github.com/microsoft/vscode/issues/183890) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-05-31 02:45:06
  10. 💪 Opened PR [#183507](https://github.com/microsoft/vscode/pull/183507) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-05-26 08:40:55
  11. ❗ Opened issue [#183506](https://github.com/microsoft/vscode/issues/183506) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-05-26 08:40:37
  
  ### [microsoft/vscode-l10n](https://github.com/microsoft/vscode-l10n)
  
  1. 💪 Opened PR [#120](https://github.com/microsoft/vscode-l10n/pull/120) in [microsoft/vscode-l10n](https://github.com/microsoft/vscode-l10n) at 2023-07-07 10:08:56
  2. ❗ Opened issue [#119](https://github.com/microsoft/vscode-l10n/issues/119) in [microsoft/vscode-l10n](https://github.com/microsoft/vscode-l10n) at 2023-07-07 09:33:51
  3. 💪 Opened PR [#118](https://github.com/microsoft/vscode-l10n/pull/118) in [microsoft/vscode-l10n](https://github.com/microsoft/vscode-l10n) at 2023-06-30 07:22:05
  
  ### [microsoft/vscode-test-web](https://github.com/microsoft/vscode-test-web)
  
  1. 💪 Opened PR [#86](https://github.com/microsoft/vscode-test-web/pull/86) in [microsoft/vscode-test-web](https://github.com/microsoft/vscode-test-web) at 2023-06-27 08:48:30
  
  ## harbin1053020115's activities:
  
  ### [microsoft/vscode](https://github.com/microsoft/vscode)
  
  1. 💪 Opened PR [#182950](https://github.com/microsoft/vscode/pull/182950) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-05-19 08:08:25
  2. ❗ Opened issue [#182949](https://github.com/microsoft/vscode/issues/182949) in [microsoft/vscode](https://github.com/microsoft/vscode) at 2023-05-19 08:03:13`.split(
    "\n"
  );
  const readmeContent = fs.readFileSync(`./README.md`, "utf-8").split("\n");

  // Find the index corresponding to <!--START_SECTION:activity--> comment
  let startIdx = readmeContent.findIndex(
    (content) => content.trim() === "<!--START_SECTION:activity-->"
  );

  // Early return in case the <!--START_SECTION:activity--> comment was not found
  if (startIdx === -1) {
    return tools.exit.failure(
      `Couldn't find the <!--START_SECTION:activity--> comment. Exiting!`
    );
  }

  // Find the index corresponding to <!--END_SECTION:activity--> comment
  let endIdx = readmeContent.findIndex(
    (content) => content.trim() === "<!--END_SECTION:activity-->"
  );

  console.log(`startIdx: ${startIdx}, endIdx: ${endIdx}`);

  if (startIdx !== -1 && endIdx === -1) {
    // Add one since the content needs to be inserted just after the initial comment
    startIdx++;
    content.forEach(
      (line, idx) => readmeContent.splice(startIdx + idx, 0, line) // 格式在前面处理
    );

    // Append <!--END_SECTION:activity--> comment
    readmeContent.splice(
      startIdx + content.length,
      0,
      "<!--END_SECTION:activity-->"
    );

    // Update README
    fs.writeFileSync(`./README.md`, readmeContent.join("\n"));

    console.log(`writeFileSync: ${readmeContent.length}`);
  }

  const oldContent = readmeContent.slice(startIdx + 1, endIdx).join("\n");
  const newContent = content.join("\n");

  if (oldContent.trim() === newContent.trim()) {
    console.log("No changes detected");
  }

  startIdx++;

  // Recent GitHub Activity content between the comments
  const readmeActivitySection = readmeContent.slice(startIdx, endIdx);
  if (readmeActivitySection.length) {
    // 清除旧数据
    readmeContent.splice(startIdx, startIdx + readmeActivitySection.length);
    endIdx = readmeContent.findIndex(
      (content) => content.trim() === "<!--END_SECTION:activity-->"
    );
  }

  content.forEach((line, idx) => {
    readmeContent.splice(startIdx + idx, 0, line);
  });
  console.log(`Wrote to ${TARGET_FILE}`);

  // Update README
  fs.writeFileSync(`./${TARGET_FILE}`, readmeContent.join("\n"));
  console.log(`writeFileSync: ${readmeContent.length}`);
  console.log(`${readmeContent.join("\n")}`);
})();
